mlflow/llama_index/serialize_objects.py

```diff
@@ -0,0 +1,224 @@
+import importlib
+import inspect
+import json
+import logging
+from typing import Callable, Dict, Set, Tuple
+
+from llama_index.core import PromptTemplate, Settings
+
+_logger = logging.getLogger(__name__)
+
+_SUPPORTED_SERIALIZATION_METHODS = ["to_dict", "dict"]
+
+
+def _extract_constructor_from_object(o: object) -> Callable:
+    if hasattr(o, "__class__"):
+        for cls in inspect.getmro(o.__class__):
+            if hasattr(cls, "__init__"):
+                return cls
+        raise AttributeError(
+            f"Class {o.__class} does not have a __init__ method and thereby cannot "
+            "be concerted to a callable."
+        )
+    else:
+        raise AttributeError(
+            f"Object {o} does not have __class__ attribute and thereby cannot be "
+            "converted to callable."
+        )
+
+
+def _get_dict_method_if_exists(o: object) -> Dict[str, any]:
```

Comment:
Let's change this to isinstance(BaseClass) for the types that we are certain can be serialized :)